decidim-proposals/app/controllers/concerns/decidim/proposals/orderable.rb

```diff
@@ -16,18 +16,51 @@ module Orderable
         # Available orders based on enabled settings
         def available_orders
           @available_orders ||= begin
-            available_orders = %w(random recent)
-            available_orders << "most_voted" if most_voted_order_available?
-            available_orders << "most_endorsed" if current_settings.endorsements_enabled?
-            available_orders << "most_commented" if component_settings.comments_enabled?
-            available_orders << "most_followed" << "with_more_authors"
+            available_orders = [default_order]
+            available_orders.concat(possible_orders.reject { |order| order == default_order })
             available_orders
           end
         end
 
+        def possible_orders
+          @possible_orders ||= begin
+            possible_orders = %w(random recent)
+            possible_orders << "most_voted" if most_voted_order_available?
+            possible_orders << "most_endorsed" if current_settings.endorsements_enabled?
+            possible_orders << "most_commented" if component_settings.comments_enabled?
+            possible_orders << "most_followed" << "with_more_authors"
+            possible_orders
+          end
+        end
+
         def default_order
+          @default_order ||= begin
+            if active_step
+              step_default_sort_order = current_component[:settings].dig("steps", active_step.id.to_s, "default_sort_order")
+              if step_default_sort_order.present?
+                return order_by_default if step_default_sort_order == "default"
+
+                return step_default_sort_order if possible_orders.include?(step_default_sort_order)
+              end
+            end
+
+            return component_settings.default_sort_order if component_settings.respond_to?(:default_sort_order) &&
```

Comment:
Do we need to check `respond_to?` as the setting should be always available when it's defined for the component (and step settings as well)?